Userland/Libraries/LibGfx/ColorSpace.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2024, Lucas Chollet <lucas.chollet@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Error.h>
+#include <AK/Noncopyable.h>
+#include <LibIPC/Forward.h>
+
+#include <core/SkColorSpace.h>
```

Comment:
Let's try to avoid skia includes in public headers (is this also the reason skia is declared a public linkage in the cmake?)

We try to keep dependencies used within .cpp files to avoid leaking their includes, so the whole world doesn't need to know about skia things. See LibWeb/Painting/DisplayListPlayerSkia.cpp as an example of how we keep this contained.

Reply:
> Let's try to avoid skia includes in public headers (is this also the reason skia is declared a public linkage in the cmake?)

Yes it is.

I'm not sure how I can keep that separated though. Let's say I have an opaque type `SkColorSpaceWrapper`, I put a forward declaration in `ColorSpace.h` and a full definition in `ColorSpace.cpp`. But then, how do I get the `sk_sp<SkColorSpace>` out of this wrapper in `DisplayListPlayerSkia` when I need to use it? I can't find a solution without hacks.

Reply:
Yeah it's a bit awkward. One semi-hacky thing I thought of is to declare the `color_space` getter as a template, and fully specialize it with a Skia definition in the .cpp. Then DisplayListPlaySkia can provide the Skia type since it knows about those types.

Something like (warning: I did not put any effort into making this maximally pretty 🙃 ):
https://github.com/trflynn89/ladybird/commit/2258c6ad38e9dfd174a2cb37aa7c53204c85ecb8

Reply:
Ok so I moved every skia related to `ColorSpaceImpl`, so `ColorSpace.h` is now a fine public header. It also means that I was able to remove the cmake modifications that I made.
However, because of the problem mentioned in my previous comment, we still need to access to the internal of `ColorSpaceImpl` from two cpp files, so I put its definition in a separated `.h`. We just need to make sure it is only included from within `LibGfx`. Is that solution fine with you?

Reply:
I've finally done something close to what you suggested.